File: src/main/java/xylixlike/utils/FXUtils.java
package xylixlike.utils;

import javafx.event.ActionEvent;
import javafx.event.EventHandler;
import javafx.scene.control.Button;


public class FXUtils {
    private FXUtils() {
        throw new IllegalStateException("Utility class");
    }

    public static Button createButton(String name, EventHandler<ActionEvent> action) {
        Button button = new Button(name);
        button.setOnAction(action);
        button.setId(name);
        return button;
    }
}

File: src/main/kotlin/xylixlike/utils/FxUtils.kt
package xylixlike.utils

import javafx.event.ActionEvent
import javafx.event.EventHandler
import javafx.scene.control.Button

fun createButton(name: String, action: EventHandler<ActionEvent>): Button {
    val button = Button(name)
    button.onAction = action
    button.id = name
    return button
}
